src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -184,10 +184,22 @@ double FFN<OutputLayerType, InitializationRuleType>::Evaluate(
   currentTarget = responses.unsafe_col(i);
 
   Forward(std::move(currentInput));
-  double res = outputLayer.Forward(std::move(boost::apply_visitor(
+  currentCost = outputLayer.Forward(std::move(boost::apply_visitor(
       outputParameterVisitor, network.back())), std::move(currentTarget));
 
-  return res;
+  return currentCost;
+}
+
+template<typename OutputLayerType, typename InitializationRuleType>
+double FFN<OutputLayerType, InitializationRuleType>::Evaluate(
```

Comment:
I am not sure I am getting the problem. Maybe I am missing something. These return conditions are above both Gradient call and Evaluate call in the loop. So I think the overallObjective is correct at that point for returning even if the number of functions is 1.

and I have realized another mistake. If we run out of iterations for ANN. The last loop which computes overall Objective again has only Evaluate without gradient call. Thus all the Evaluate will return the same objective. I should check for objective function index there.